src/autoloads/scenes.gd

```diff
@@ -0,0 +1,133 @@
+#
+## back() -> void
+# Goes back a scene, based on the order in sceneOrder. The function will instead call the function "_onBack"
+# on the base scene, if the current scene matches lowestScenePath. The lowestScenePath is always at the front
+# of sceneOrder.
+#
+## setBase(scene: Node, lowest: String) -> void
+# Sets the base scene to a currently loaded scene. Should only be used if baseScene is unset. Lowest defines
+# the path to set lowestScenePath.
+# This function is specifically for setting the baseScene from a scene already instanced in the editor. After
+# baseScene is set, switchBase() should be called instead to change the base scene.
+#
+## switchBase(path: String, lowest: String) -> void
+# Switches out the base scene for the one defined in path. Lowest defines the path to set lowestScenePath.
+#
+## overlay(path: String) -> void
+# Adds the scene specified by it's resource path, as a child of the base scene. All other child scenes are 
+# hidden, and the newly overlayed scene is shown. If the scene is already loaded, it is just shown instead. 
+# The _focus method, if it exists, is run on the scene.
+#
+## preloadOverlay(path: String) -> void
+# Similar to overlay(), except the scene is loaded in the background, and not shown. The _focus method is not
+# run on the scene either.
+
+var baseScene: Node # Always loaded scene
+var baseScenePath: String # Path of baseScene
+
+var lowestScenePath: String # Path to scene that won't be hidden by back
+
+var loadedScenes: Dictionary # Loaded scenes with their paths as keys
+var sceneOrder: Array # Order of scenes, with back being the top
+
+# --Public Functions--
+
+# Add a child scene to the base scene, and show it
+func overlay(path: String) -> void:
+	_hideAllLoaded() # Hide all loaded overlays
+	_checkLowestPath() # Check that the lowest path is in the sceneOrder array
+	if loadedScenes.has(path):
+		_showLoaded(path) # Show the scene if it has already been loaded
+	else:
+		call_deferred("_deferredOverlay", path, true) # Otherwise load the scene
+	if path != sceneOrder.back():
+		sceneOrder.append(path) # Add the path to sceneOrder if it isn't already there
+
+# Preload a child scene to the base scene, but don't show it
+func preloadOverlay(path: String) -> void:
+	if not loadedScenes.has(path):
+		call_deferred("_deferredOverlay", path, false) # Load the scene if not already loaded
+
+# Set a new scene to be the base scene
+func switchBase(path: String, lowest: String) -> void:
+	loadedScenes.clear()
+	sceneOrder.clear()
+	lowestScenePath = lowest
+	call_deferred("_deferredSwitchBase", path) # Switch out the base scene
+
+# Set the base scene from an already loaded scene. Only to be used if base scene hasn't been set.
+func setBase(scene: Node, lowest: String) -> void:
+	assert(baseScene == null, "Please use switchBase instead.")
+	baseScene = scene
+	baseScenePath = scene.filename
+	loadedScenes[baseScenePath] = baseScene
+	lowestScenePath = lowest
+	overlay(lowestScenePath)
```

Comment:
If I'm not mistaken, this function is only called at the start of the game.
I think that the "lowest" parameter is confusing, and also probably not needed (especially it being copied into lowestScenePath). The overlay function called from here could be more cleanly called from...